pkg/ipam/allocator/podcidr/podcidr.go

```diff
@@ -290,10 +290,8 @@ func syncToK8s(nodeGetterUpdater ipam.CiliumNodeGetterUpdater, ciliumNodesToK8s
 				err = nil
 			}
 		case k8sOpDelete:
-			err = nodeGetterUpdater.Delete(nodeName)
-			if k8sErrors.IsNotFound(err) || k8sErrors.IsGone(err) {
-				err = nil
-			}
```

Comment:
We still have a case where we rely on this code at https://github.com/cilium/cilium/blob/248831f3027f5d9fb26a5ce85f67b6789171a055/pkg/ipam/allocator/podcidr/podcidr.go#L406-L409

If I understand correctly, we should remove that code as well because we are now only deleting nodes from `nodeManager` in reaction to `CiliumNode` delete events, which are triggered by `Node` delete events. So no need to mark the node to deleted in k8s when we remove from `nodeManager`. Does that make sense?

Do we basically don't need `k8sOpDelete` at all anymore?